#### 예외처리

코틀린의 예외처리는 자바나 다른 언어의 예외처리와 비슷하다.  정상적으로 종료되지 않을 경우 예외를 던질 수 있다. 함수를 호출하는 쪽에서는 그 예외를 잡아 처리할 수 있다.

if (){

throw {

}}

예외 인스턴스를 만들때 new를 붙일 필요가없다. 하지만 자바와 다르게 throw는 식이므로 다른 식에 포함 될 수 있다.

##### try,catch,finally

예외를 처리하려면 try,catch,finally 절을 함계 사용한다. 파일에서 각 줄을 읽어 수로 변환하고 그 줄이 올바른 수 형태가 아니면 null 을 반환 할 수 있다.



fun readNumber(reader:BufferedReader):int ?{

try{

​		val line = reader.readLine()

​		return Integer.parseInt(line)

}

catch(e:NumberFormatException){}

finally{}

}

자바코드와 가장 큰 차이는 throws절이 없다는 것이다. 자바에서는 함수를 작성할떄 함수선언뒤 체크예외가 되기 때문에 throws IoException을 붙여야한다.  하지만 코틀린에서는 함수가 던지는 예외를 지정하지 않고 발생한 예외를 잡아도,안잡아도 크게 상관이없다.